File: macos-reverse-engineering-10weeks/code/week08/ioc_report.py
#!/usr/bin/env python3
# ============================================================
# ioc_report.py — Sinh khung báo cáo IOC (Indicators of Compromise)
# Mục tiêu: tạo template báo cáo IOC để điền kết quả phân tích.
# Đầu vào: không.
# Đầu ra: khung báo cáo IOC (in ra).
# An toàn: dữ liệu do bạn điền; không chứa secret/mẫu thật.
# ============================================================

def main():
    print("""# Báo cáo IOC — macOS

## 1. Nhận dạng mẫu
- Hash (SHA-256): <hash>
- Kiến trúc / định dạng: <arm64 / Mach-O>
- Tên file: <name>

## 2. Static analysis
- Strings đáng chú ý: <URL, key, path>
- Symbol / class: <class-dump, nm>
- Entitlements / code signing: <codesign output>

## 3. Dynamic analysis
- File ops: <fs_usage>
- Process: <tiến trình tạo ra>
- Network: <host, port>

## 4. Persistence
- LaunchAgents: <plist path>
- Login items / cron: <nếu có>

## 5. Indicators of Compromise
| Loại | Giá trị |
|---|---|
| File | <path, hash> |
| Network | <domain, IP, port> |
| Plist | <label, path> |

## 6. Kết luận và phòng thủ
<nhận định + biện pháp>
""")


if __name__ == "__main__":
    main()

File: macos-reverse-engineering-10weeks/code/week08/ioc_report.cpp
// 08_ioc_report.cpp — Sinh báo cáo IOC macOS (persistence + mạng + file)
// Mục tiêu: tổng hợp dấu hiệu xâm nhập (LaunchAgent, file, domain) thành báo cáo.
// An toàn: thuần tính toán, chỉ in văn bản.  g++ -std=c++17 -O2 ioc_report.cpp -o demo
#include <iostream>
#include <string>
#include <vector>

int main() {
    std::vector<std::pair<std::string,std::string>> iocs = {
        {"LaunchAgent", "~/Library/LaunchAgents/com.attacker.agent.plist"},
        {"File",        "/tmp/payload.dylib"},
        {"Domain",      "evil.example"},
        {"Hash",        "7f83b1657ff1fc53"},
    };
    std::cout << "BÁO CÁO IOC — macOS\n" << std::string(56,'=') << "\n\n";
    for (auto& [loai, giatri] : iocs) std::cout << "  [" << loai << "] " << giatri << "\n";
    std::cout << "\nKhắc phục: xoá LaunchAgent, chặn domain, xoá snapshot VM.\n";
    return 0;
}
